### ISSUE TITLE
ServiceBinding for Kubernetes

### ISSUE THREAD

**Issue by corneil** [spring-cloud/spring-cloud-kubernetes#1248]
**Is your feature request related to a problem? Please describe.**
Connecting to external backing services like database and message brokers and even common services requires credentials and multiple properties.

**Solution description:**
ServiceBindings for Kubernetes specification 1.0.0 has been release and is worth exploring.

Hopefully the specification leverages existing functionality in a way that reduces impact.

**Additional context**
It may come down to a starter and a property that enables the service-bindings and augments existing service discovery, propertysource.

**Comment by scottfrederick** [comment]
@corneil The [Spring Cloud Bindings](https://github.com/spring-cloud/spring-cloud-bindings) project provides the ability to map from Kubernetes service bindings to Spring Boot connection properties. Does that project provide the features you're thinking of?

**Comment by corneil** [comment]
@scottfrederick Yes. Now I'm wondering why isn't  is part of Spring Cloud Kubernetes?

**Comment by wind57** [comment]
I don't think it should be part of spring-cloud-kubernetes, since that spring-cloud-bindings is supposed to do that across all cloud based projects; this is how I see it. if that is the case indeed, then this is safe to close. @ryanjbaxter
